src/gui/redux/docgen/docgen.api.js

```diff
@@ -0,0 +1,16 @@
+import docgen from './docgen'
+
+export const fetchDocgen = () => async (dispatch) => {
+  dispatch(docgen.reset())
+  try {
+    const response = await fetch('/api/doc')
+    const json = await response.json()
+    if (json && json.length > 0) {
+      dispatch(docgen.set(json[0]))
+    } else {
+      dispatch(docgen.set({}))
+    }
+  } catch (error) {
+    dispatch(docgen.set({}))
```

Comment:
To manage errors, should I add a middleware ? or create a dedicated reducer for errors ? or redux-factory should manage errors ?